presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/optimization/JdbcComputePushdown.java

```diff
@@ -28,24 +30,41 @@
 import com.facebook.presto.spi.plan.TableScanNode;
 import com.facebook.presto.spi.relation.DeterminismEvaluator;
 import com.facebook.presto.spi.relation.ExpressionOptimizer;
+import com.facebook.presto.spi.relation.RowExpression;
 import com.google.common.collect.ImmutableList;
 
 import java.util.Optional;
+import java.util.Set;
 
+import static com.facebook.presto.expressions.translator.FunctionTranslator.buildFunctionTranslator;
+import static com.facebook.presto.expressions.translator.RowExpressionTreeTranslator.translateWith;
+import static com.facebook.presto.spi.relation.ExpressionOptimizer.Level.OPTIMIZED;
 import static java.util.Objects.requireNonNull;
 
 public class JdbcComputePushdown
         implements ConnectorPlanOptimizer
 {
     private final ExpressionOptimizer expressionOptimizer;
+    private final JdbcFilterToSqlTranslator jdbcFilterToSqlTranslator;
+    private final LogicalRowExpressions logicalRowExpressions;
 
     public JdbcComputePushdown(
             FunctionMetadataManager functionMetadataManager,
             StandardFunctionResolution functionResolution,
             DeterminismEvaluator determinismEvaluator,
-            ExpressionOptimizer expressionOptimizer)
+            ExpressionOptimizer expressionOptimizer,
+            String identifierQuote,
+            Set<Class<?>> functionTranslators)
     {
-        this.expressionOptimizer = expressionOptimizer;
+        this.expressionOptimizer = requireNonNull(expressionOptimizer);
```

Comment:
We usually have a list of requireNonNull checks with error messages and then assign them without requireNonNull if they will be reused multiple times.
```java
requireNonNull(functionMetadataManager, "functionMetadataManager is null");
requireNonNull(...);
requireNonNull(...);
...

this.jdbcFilterToSqlTranslator = new JdbcFilterToSqlTranslator(functionMetadataManager, buildFunctionTranslator(functionTranslators), identifierQuote);
this.logicalRowExpressions = new LogicalRowExpressions(...);
```

or 

```java
this.functionMetadataManager = requireNonNull(functionMetadataManager, "functionMetadataManager is null");